src/main/java/org/jenkinsci/plugins/github_branch_source/ExcludeArchivedTrait.java

```diff
@@ -0,0 +1,54 @@
+package org.jenkinsci.plugins.github_branch_source;
+
+import hudson.Extension;
+import jenkins.scm.api.trait.SCMNavigatorContext;
+import jenkins.scm.api.trait.SCMNavigatorTrait;
+import jenkins.scm.api.trait.SCMNavigatorTraitDescriptor;
+import jenkins.scm.impl.trait.Selection;
+import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import javax.annotation.Nonnull;
+
+/**
+ * A {@link Selection} trait that will restrict the discovery of repositories that have been archived.
+ */
+public class ExcludeArchivedTrait extends SCMNavigatorTrait {
+
+    /**
+     * Constructor for stapler.
+     */
+    @DataBoundConstructor
+    public ExcludeArchivedTrait() {
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void decorateContext(SCMNavigatorContext<?, ?> context) {
+        super.decorateContext(context);
+        GitHubSCMNavigatorContext ctx = (GitHubSCMNavigatorContext) context;
+        ctx.setExcludeArchived(true);
+    }
+
+    /**
+     * Excluded archived repositories filter
+     */
+    @Symbol("archivedRepositoriesFilter")
```

Comment:
Similar to a number of other traits, this will likely migrate to other SCMs as well and adding `gitHub` to the front will prevent later breaking changes. 
And let's make all the names line up with each other.  `ExcludeArchivedRepositories` is descriptive and unambiguous. 
(I'm open to disagreement and discussion on the specific name.) 
```suggestion
    @Symbol("gitHubExcludeArchivedRepositories")
```

Reply:
That sounds like a good idea to me.